[Bug Report] Update error

## WHAT

When updating a package that is already installed, there is an error that appears to be a problem with the timing of removing the past. When this happens, it is inconvenient because it requires re-entry of GITHUB_TOKEN.

```sh
❯ afx update
? OK to update these packages? cli/cli Yes
✖ cli/cli
[ERROR]: 1 error occurred:
        * 1 error occurred:
                * cli/cli: failed to get command.link: cli/cli: 2 files matched: []string{"/home/uga/.afx/github.com/cli/cli/gh_2.14.6_linux_amd64/bin/gh", "/home/uga/.afx/github.com/cli/cli/gh_2.14.4_linux_amd64/bin/gh"}
```

Here is an excerpt from my release.yaml.
```yaml
- github:
  - name: cli/cli
    description: Github's official command line tool
    owner: cli
    repo: cli
    release:
      name: gh
      tag: v2.14.6
      asset:
        filename: '{{ .Release.Name }}_{{ replace .Release.Tag "v" "" }}_{{ .OS }}_{{ .Arch }}.tar.gz'
    command:
      link:
        - from: '**/gh'
```

## WHY

Bug report.

```sh
$ AFX_LOG=debug afx update
2023/03/10 22:58:36 [INFO] afx version: 0.1.25
2023/03/10 22:58:36 [INFO] Go runtime version: go1.16.1
2023/03/10 22:58:36 [INFO] Build tag/SHA: v0.1.25/e9d6a7d
2023/03/10 22:58:36 [INFO] CLI args: []string{"afx", "update"}
2023/03/10 22:58:36 [INFO] Reading config /home/uga/.config/afx/github.yaml...
2023/03/10 22:58:36 [DEBUG] (goroutine): checking new updates...
2023/03/10 22:58:36 [INFO] Parsing config...
2023/03/10 22:58:36 [INFO] Reading config /home/uga/.config/afx/main.yaml...
2023/03/10 22:58:36 [INFO] Parsing config...
2023/03/10 22:58:36 [INFO] Reading config /home/uga/.config/afx/release.yaml...
2023/03/10 22:58:36 [INFO] Parsing config...
2023/03/10 22:58:36 [DEBUG] dependency graph is here:
* sindresorhus/pure -> mafredri/zsh-async
2023/03/10 22:58:36 [DEBUG] mkdir /home/uga/.afx
2023/03/10 22:58:36 [DEBUG] mkdir /home/uga/.afx/bin
2023/03/10 22:58:36 [ERROR] there're some states or packages which needs operations: cannot refresh state
2023/03/10 22:58:36 [INFO] state additions: (0) []string(nil)
2023/03/10 22:58:36 [INFO] state deletions: (0) []string(nil)
2023/03/10 22:58:36 [INFO] state changes: (1) []string{"sharkdp/fd"}
2023/03/10 22:58:36 [INFO] state unchanges: (17) []string{...skip...}
? OK to update these packages? sharkdp/fd Yes
2023/03/10 22:58:38 [DEBUG] (update): start to run each pkg.Install()
2023/03/10 22:58:38 [DEBUG] asset: templating filename from "{{ .Release.Name }}-{{ .Release.Tag }}-{{ .Arch }}-{{ .OS }}.tar.gz"
2023/03/10 22:58:38 [DEBUG] asset: templated filename: -> "fd-v8.6.0-x86_64-unknown-linux-musl.tar.gz"
2023/03/10 22:58:38 [DEBUG] getting asset data from https://api.github.com/repos/sharkdp/fd/releases/tags/v8.6.0
2023/03/10 22:58:38 [DEBUG] assets: []string{"fd-musl_8.6.0_amd64.deb", "fd-musl_8.6.0_i686.deb", "fd-v8.6.0-aarch64-unknown-linux-gnu.tar.gz", "fd-v8.6.0-arm-unknown-linux-gnueabihf.tar.gz", "fd-v8.6.0-arm-unknown-linux-musleabihf.tar.gz", "fd-v8.6.0-i686-pc-windows-msvc.zip", "fd-v8.6.0-i686-unknown-linux-gnu.tar.gz", "fd-v8.6.0-i686-unknown-linux-musl.tar.gz", "fd-v8.6.0-x86_64-apple-darwin.tar.gz", "fd-v8.6.0-x86_64-pc-windows-gnu.zip", "fd-v8.6.0-x86_64-pc-windows-msvc.zip", "fd-v8.6.0-x86_64-unknown-linux-gnu.tar.gz", "fd-v8.6.0-x86_64-unknown-linux-musl.tar.gz", "fd_8.6.0_amd64.deb", "fd_8.6.0_arm64.deb", "fd_8.6.0_armhf.deb", "fd_8.6.0_i686.deb"}
2023/03/10 22:58:38 [DEBUG] asset: filterfunc: started running
2023/03/10 22:58:38 [DEBUG] asset: filterfunc: matched in assets
2023/03/10 22:58:38 [DEBUG] asset: github.Asset{Name:"fd-v8.6.0-x86_64-unknown-linux-musl.tar.gz", URL:"https://github.com/sharkdp/fd/releases/download/v8.6.0/fd-v8.6.0-x86_64-unknown-linux-musl.tar.gz"}
2023/03/10 22:58:42 [DEBUG] uaIface: &archiver.TarGz{Tar:(*archiver.Tar)(0xc000387080), CompressionLevel:-1} (*archiver.TarGz)
2023/03/10 22:58:42 [DEBUG] removed archive file: /home/uga/.afx/github.com/sharkdp/fd/fd-v8.6.0-x86_64-unknown-linux-musl.tar.gz
2023/03/10 22:58:42 [DEBUG] uninstall "sharkdp/fd" because updating failed
✖ sharkdp/fd
2023/03/10 22:58:42 [INFO] Delete /home/uga/.afx/github.com/sharkdp/fd
2023/03/10 22:58:42 [INFO] root command execution finished
[ERROR]: 1 error occurred:
        * 1 error occurred:
                * sharkdp/fd: failed to get command.link: sharkdp/fd: 2 files matched: []string{"/home/uga/.afx/github.com/sharkdp/fd/fd-v8.7.0-x86_64-unknown-linux-musl/fd", "/home/uga/.afx/github.com/sharkdp/fd/fd-v8.6.0-x86_64-unknown-linux-musl/fd"}
```
The error is in the install that is called inside the update, I think it can be fixed by deleting the old version of the directory before pasting the symlink.

--- a/cmd/update.go
+++ b/cmd/update.go
@@ -114,6 +114,7 @@ func (c *updateCmd) run(pkgs []config.Package) error {
 		eg.Go(func() error {
 			limit <- struct{}{}
 			defer func() { <-limit }()
+			os.RemoveAll(pkg.GetHome()) // delete before updating
 			err := pkg.Install(ctx, completion)
 			switch err {
 			case nil:

